source/dem/velocity_verlet_integrator.cc

```diff
@@ -141,12 +141,148 @@ VelocityVerletIntegrator<dim>::integrate(
+                              {
+                                return particle.get_location();
+                              }
+                            else
+                              {
+                                return (
+                                  point_nd_to_3d(particle.get_location()));
+                              }
+                          }();
+
+                          for (unsigned int d = 0; d < 3; ++d)
+                            {
+                              // Particle velocity integration
+                              particle_properties[PropertiesIndex::v_x + d] +=
+                                dt_g[d] + particle_force[d] * dt_mass_inverse;
+
+                              // Particle location integration
+                              particle_position[d] +=
+                                particle_properties[PropertiesIndex::v_x + d] *
+                                dt;
+
+                              // Updating angular velocity
+                              particle_properties[PropertiesIndex::omega_x +
+                                                  d] +=
+                                particle_torque[d] * dt_MOI_inverse;
+                            }
+
+                          // Reinitialize force and torque of particle
+                          particle_force  = 0.0;
+                          particle_torque = 0.0;
+
+                          // Update particle location
+                          if constexpr (dim == 3)
+                            particle.set_location(particle_position);
+
+                          if constexpr (dim == 2)
+                            {
+                              Point<2> position_2d;
+                              position_2d[0] = particle_position[0];
+                              position_2d[1] = particle_position[1];
+                              particle.set_location(position_2d);
+                            }
+                        }
+                    }
+                  else // Active and inactive cells
+                    {
+                      for (auto &particle : particles_in_cell)
+                        {
+                          types::particle_index particle_id =
+                            particle.get_local_index();
+
+                          // Reset forces
+                          force[particle_id]  = 0.0;
+                          torque[particle_id] = 0.0;
+                        }
+                    }
+                }
+            }
+        }
+    }
+  else
+    {
+      integrate_with_advected_particles(particle_handler,
+                                        g,
+                                        dt,
+                                        torque,
+                                        force,
+                                        MOI,
+                                        triangulation,
+                                        sparse_contacts_object);
+    }
```

Comment:
I know it might sound weird, but I would suggest to put this into a if instead and then continue with the rest of the code.
This way you don't have two huge blocks with the logic ecompassed
You check and you call, otherwise you do the main thing. I often think it's easier to understand if you have a small block then an else than a huuuuge block and a small else.